iothub/service/src/Common/TrackingHelper.cs

```diff
@@ -16,7 +16,9 @@ namespace Microsoft.Azure.Devices.Common
 
     public static class TrackingHelper
     {
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Usage", "CA2211:Non-constant fields should not be visible", Justification = "This property may be used by others so it is not safe to modify it.")]
         public static string GatewayId;
```

Comment:
Where's the obsolete attribute? Are we going to add a property with a private setter to replace it?

Reply:
No, there would be no need. 

If we add a private setter we'll still need to have some public mutator method that just sets the value to something since the idea is this field is to be reused across many threads.